xbmc/guilib/GUIInfoTypes.h

```diff
@@ -125,6 +126,27 @@ class CGUIInfoLabel
    */
   static std::string ReplaceAddonStrings(const std::string &label);
 
+  typedef boost::function<std::string(const std::string&)> StringReplacerFunc;
+
+  /*!
+   \brief Replaces instances of $strKeyword[value] with the appropriate resolved string
+   \param strInput text to replace
+   \param strKeyword keyword to look for
+   \param func function that does the actual replacement of each bracketed value found
+   \param strOutput the output string
+   \return whether anything has been replaced.
+   */
+  static bool ReplaceDollarKeywordStrings(const std::string &strInput, const std::string &strKeyword, const StringReplacerFunc &func, std::string &strOutput);
```

Comment:
Parameter references are not resolved in a "global" context and are not really "info" labels. They just happen to use the same syntax: `$KEYWORD[]`. IMO, `ReplaceStrings` name is not descriptive enough. It sounds like its doing the same job as `std::replace()`, while in reality it does something vastly different. Perhaps we can just think of a better (but descriptive) name for `ReplaceDollarKeywordStrings()`? I am open to suggestions.

Btw, if you check out early posts on the discussion page, you'll see that I originally wanted to put `ReplaceDollarKeywordStrings()` into some kind of a util class, but simply didn't find the appropriate one. I think you probably wouldn't mind the design if I've done that. My point is that this method should not be exclusively tied to addon/info labels, as it is more general now. It parses all `$KEYWORD[]` strings, whatever they happen to mean.

Perhaps an analogy could help. Moving all (unrelated) replacer functors into the same file with the *util* method `ReplaceDollarKeywordStrings()` would be equal to, say, bundling all comparison functors you used in your program together with `std::sort()` directly in `<algorithm>` header. That's probably not that good of a design. Some of them perhaps need to be friends with your classes etc. and there are also compilation dependency issues etc.